components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -4114,6 +4114,58 @@ public String getServiceProviderNameByClientId(String clientId, String clientTyp
         return applicationName;
     }
 
+    /**
+     * @param inboundKey inbound key
+     * @param inboundType inbound type
+     * @param tenantDomain tenant domain
+     * @return application resource id
+     * @throws IdentityApplicationManagementException
+     */
+    public String getApplicationResourceIDByInboundKey(String inboundKey, String inboundType, String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        int tenantID = MultitenantConstants.SUPER_TENANT_ID;
+
+        if (StringUtils.isEmpty(inboundKey)) {
+            return null;
+        }
+
+        if (tenantDomain != null) {
+            try {
+                tenantID = ApplicationManagementServiceComponentHolder.getInstance().getRealmService()
+                        .getTenantManager().getTenantId(tenantDomain);
+            } catch (UserStoreException e1) {
+                throw new IdentityApplicationManagementException("Error while reading application", e1);
+            }
+        }
+        String applicationResourceId = null;
+
+        // Reading application resource id from the database
+        Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+        PreparedStatement storeAppPrepStmt = null;
+        ResultSet appIdResult = null;
+        try {
```

Comment:
Can't we use try with resource approach here?

Reply:
fixed